plugins/optimization-detective/class-od-html-tag-walker.php

```diff
@@ -187,8 +204,15 @@ public function __construct( string $html ) {
 	 * @since 0.1.0
 	 *
 	 * @return Generator<string> Tag name of current open tag.
+	 *
+	 * @throws Exception When walking has already started.
 	 */
 	public function open_tags(): Generator {
+		if ( $this->did_start_walking ) {
+			throw new Exception( esc_html__( 'Open tags may only be iterated over once per instance.', 'optimization-detective' ) );
+		}
+		$this->did_start_walking = true;
```

Comment:
Since `WP_HTML_Tag_Processor` doesn't rewind after iterating over tags in the document, it's important that `open_tags()` also not support being invoked multiple times to walk over the document.